### PR TITLE
Don't run workflows when code hasn't changed

### PR DESCRIPTION
**Pull request by nirinchev**
## Description

Fixes https://github.com/realm/realm-dotnet/issues/2850

### CODE REVIEW

**Review by fealebenpae — COMMENTED** [review]
Could we use `Realm/**` and `wrappers/**` for paths instead? Also, do you know if this will be triggered for PRs with just the realm-core submodule changed?

**Comment by nirinchev** [comment]
@fealebenpae now it should run if we update realm-core. I wanted to be more explicit about the monitored types because we do have some metadata/docs files in `Realm`, `Tests`, and so on and I'd like to avoid running the PR flow for those. I went through ~10-15 merged PRs and this filter would have done the right thing for them.

File: .github/templates/main.yml [hunk 1]
```diff
@@ -413,6 +413,15 @@ name: main
     branches:
       - main
   pull_request:
+    paths:
+      - '**.cs'
+      - '**.cpp'
+      - '**.hpp'
+      - '**.csproj'
```
Comment (fealebenpae) [inline .github/templates/main.yml:420]:
```suggestion
      - '**.csproj'
      - '*CMakeLists.txt'
```

**Review by fealebenpae — APPROVED** [review]

File: .github/templates/main.yml (hunk 1 above)
Comment (fealebenpae) [reply .github/templates/main.yml:420]: Looks good, but we should also take CMakeLists.txt files into account.

**Pull request merged**